exp/api/v1beta1/azuremanagedmachinepool_webhook.go

```diff
@@ -205,6 +210,18 @@ func (r *AzureManagedMachinePool) validateMaxPods() error {
 	return nil
 }
 
+func (r *AzureManagedMachinePool) ValidateSpotNodePool() error {
+	if r.Spec.ScaleSetPriority != nil && *r.Spec.ScaleSetPriority == "Spot" {
+		if r.Spec.Mode != string(NodePoolModeUser) {
+			return field.Invalid(
+				field.NewPath("Spec", "ScaleSetPriority"),
+				r.Spec.ScaleSetPriority,
+				"Spot ScaleSetPriority requires AzureManagedMachinePool mode User")
+		}
+	}
```

Comment:
Let's add further validation to ensure that only the literal strings "Spot" and "Regular" are allowed. If I use a value of "Special" (or some other random string) the validation should fail.

Reply:
we have open API schema validation for that